src/evaluation/retrieval.py

```diff
@@ -1,80 +1,87 @@
 import open_clip
 import numpy as np
 import torch
+from einops import rearrange
 
-
-def retrieval_evaluation(model_video, model_text, data, multicaption=False):
+def retrieval_evaluation(model_video, model_text, data):
     if type(data) == dict:
         dataloader = data["val"].dataloader
     else:
         dataloader = data
     device = "cuda" if torch.cuda.is_available() else "cpu"
     all_video_features, all_text_features = [], []
-    ground_truth = []
-    samp = 0
-
+    
     with torch.no_grad():
         for i, batch in enumerate(dataloader):
             embeddings = batch["embeddings"]
+
             toks = []
             # TODO: does this require batch_size = 1 ??
             for cap in batch["text"]:
-                if multicaption:
-                    for c in cap.split(";"): # multiple captions separated by ;
-                        toks.append(open_clip.tokenize(c))
-                        ground_truth.append(samp)
-                else:
-                    toks.append(open_clip.tokenize(cap))
-                    ground_truth.append(samp)
-                samp += 1
+                for c in cap.split(";"): # multiple captions separated by ;
+                    toks.append(open_clip.tokenize(c))
+            
             toks = torch.cat(toks)
             embeddings = embeddings.to(device, non_blocking=True)
             toks = toks.to(device, non_blocking=True)
 
             video_embeddings = model_video(embeddings)
-            text_embeddings = model_text(toks).float()
+            text_embeddings = model_text(toks)
 
             all_video_features.append(video_embeddings.cpu())
             all_text_features.append(text_embeddings.cpu())
+        
+        video_features = torch.cat(all_video_features)
+        text_features = torch.stack(all_text_features) # maintain the 3d structure of text_features
 
         val_metrics = get_metrics(
-            video_features=torch.cat(all_video_features),
-            text_features=torch.cat(all_text_features),
-            ground_truth=ground_truth,
+            video_features=video_features,
+            text_features=text_features,
             logit_scale=100.0,
         )
-    return val_metrics
+    return val_metrics,
 
+def get_metrics(video_features, text_features, logit_scale):
+    ''' 
+    Assumptions for this eval:
 
-def get_metrics(video_features, text_features, ground_truth, logit_scale):
-    metrics = {}
+    - len(text_features) % len(video_features) == 0, i.e. there's a constant number 
```

Comment:
this assumption isn't true for MSVD (a common benchmark) for example and I'm not sure if we should have this...